instance_test.go

```diff
@@ -104,18 +105,54 @@ var _ = Describe("Instance", func() {
 				Expect(instance.Name).To(Equal("INSTTEST"), "name")
 				Expect(instance.Directory).To(Equal("/ensemble/instances/insttest/"), "directory")
 				Expect(instance.Version).To(Equal("2015.2.2.805.0.16216"), "version")
-				Expect(instance.Status).To(Equal(isclib.InstanceStatusRunning), "status")
+				Expect(instance.Status).To(Equal(InstanceStatusRunning), "status")
 				Expect(instance.Activity).To(Equal("since Fri May 13 22:07:02 2016"), "activity")
 				Expect(instance.CPFFileName).To(Equal("cache.cpf"), "cpf")
 				Expect(instance.SuperServerPort).To(Equal(56772), "ss port")
 				Expect(instance.WebServerPort).To(Equal(57772), "ws port")
 				Expect(instance.JDBCPort).To(Equal(62972), "jdbc port")
 				Expect(instance.State).To(Equal("ok"), "state")
-				Expect(instance.Product).To(Equal(isclib.Cache), "product")
+				Expect(instance.Product).To(Equal(Cache), "product")
 				Expect(instance.MirrorMemberType).To(Equal("Failover"), "mirror type")
 				Expect(instance.MirrorStatus).To(Equal("Primary"), "mirror status")
 				Expect(instance.DataDirectory).To(Equal("/mgr/config"), "data directory")
 			})
 		})
 	})
+	Context("WaitForReady", func() {
+		var i int
+		BeforeEach(func() {
+			i = 0
+		})
+		Context("With timeout", func() {
+			Context("Does not come up", func() {
+				instance, err := InstanceFromQList("INSTTEST^/ensemble/instances/insttest/^2012.2.3.903.2.12515^down, last used Thu Sep 15 18:58:30 2016^cache.cpf^56772^57772^62972")
+				ctx, can := context.WithTimeout(context.Background(), 50*time.Millisecond)
+				defer can()
+				err = instance.WaitForReady(ctx)
+				It("Returns an error", func() {
+					Expect(err).To(HaveOccurred())
+				})
+				It("Timed out", func() {
+					Expect(err).Should(MatchError(context.DeadlineExceeded))
+				})
+			})
+			Context("Does come up", func() {
+				getQlist = func(instanceName string) (string, error) {
+					if i >= 3 {
```

Comment:
I think it would be preferable if this didn't rely on knowledge of the internal refresh interval to know what this count should be.  Perhaps it would be better if you used a `time.AfterFunc` to set the value returned by `getQlist` halfway through the timeout duration?